app/screens/send-bitcoin-screen/send-bitcoin-destination-screen.tsx

```diff
@@ -111,6 +126,110 @@ const SendBitcoinDestinationScreen: React.FC<Props> = ({ route }) => {
     fetchPolicy: "no-cache",
   })
 
+  const {
+    loading,
+    data: contactsData,
+    error,
+  } = useContactsQuery({
+    skip: !isAuthed,
+    fetchPolicy: "cache-and-network",
+  })
+
+  if (error) {
+    toastShow({ message: error.message, LL })
+  }
+
+  const [matchingContacts, setMatchingContacts] = useState<Contact[]>([])
+  const allContacts: Contact[] = useMemo(() => {
+    const contactsCopy = contactsData?.me?.contacts.slice() ?? []
+    const compareUsernames = (a: Contact, b: Contact) => {
+      return a.username.toLocaleLowerCase().localeCompare(b.username.toLocaleLowerCase())
+    }
+    contactsCopy.sort(compareUsernames)
+
+    return contactsCopy
+  }, [contactsData])
+
+  const [selectedId, setSelectedId] = useState("")
+
+  const handleSelection = (id: string) => {
+    if (selectedId === id) setSelectedId("")
+    else setSelectedId(id)
+  }
+
+  const reset = useCallback(() => {
+    dispatchDestinationStateAction({
+      type: "set-unparsed-destination",
+      payload: { unparsedDestination: "" },
+    })
+    setGoToNextScreenWhenValid(false)
+    setSelectedId("")
+    setMatchingContacts(allContacts)
+  }, [allContacts])
+
+  const wordMatchesContact = (searchWord: string, contact: Contact): boolean => {
+    let contactPrettyNameMatchesSearchWord: boolean
+
+    const contactNameMatchesSearchWord = contact.username
+      .toLowerCase()
+      .includes(searchWord.toLowerCase())
+
+    if (contact.alias) {
+      contactPrettyNameMatchesSearchWord = contact.alias
+        .toLowerCase()
+        .includes(searchWord.toLowerCase())
+    } else {
+      contactPrettyNameMatchesSearchWord = false
+    }
+
+    return contactNameMatchesSearchWord || contactPrettyNameMatchesSearchWord
+  }
```

Comment:
this function doesn't need to be in the component. it would actually be better outside because it would make it easier to unit test it

Reply:
Was taken directly from [screens/people-screen/contacts/all-contacts.tsx](https://github.com/GaloyMoney/galoy-mobile/blob/84eaa80cde72b5d8fc0ad65402066ff1ed6b72d0/app/screens/people-screen/contacts/all-contacts.tsx#L53-L55) but together with the useContactsQuery. 
Removed it, let me know what you think

Reply:
Taken directly from [screens/people-screen/contacts/all-contacts.tsx](https://github.com/GaloyMoney/galoy-mobile/blob/84eaa80cde72b5d8fc0ad65402066ff1ed6b72d0/app/screens/people-screen/contacts/all-contacts.tsx#L90C3-L106C4)
If it shall be placed outside only here please confirm again :+1: